java/src/jmri/jmrix/tmcc/serialmon/SerialMonFrame.java

```diff
@@ -59,12 +59,128 @@ public synchronized void reply(SerialReply l) { // receive a reply message and l
+                        case 18:
+                            return "Engine " + A + " - Num 2 - Crew Talk";
+                        case 19:
+                            return "Engine " + A + " - Num 3 - Sound On w/Start-Up Sequence";
+                        case 20:
+                            return "Engine " + A + " - Num 4 - Sound Volume Decrease";
+                        case 21:
+                            return "Engine " + A + " - Num 5 - Sound Off w/Shut-Down Sequence";
+                        case 22:
+                            return "Engine " + A + " - Num 6 - Steam Release/RPM Decrease";
+                        case 23:
+                            return "Engine " + A + " - Num 7 - Tower Com Announcement";
+                        case 24:
+                            return "Engine " + A + " - Num 8 - Feature Off (Smoke/Aux Lighting)";
+                        case 25:
+                            return "Engine " + A + " - Num 9 - Feature On (Smoke/Aux Lighting)";
+                        case 26:
+                            
+                        case 27:
+                            
+                        case 28:
+                            return "Engine " + A + " - Blow Whistle/Horn 1";
+                        case 29:
+                            return "Engine " + A + " - Ring Bell";
+                        case 30:
+                            return "Engine " + A + " - Letoff Sound";
+                        case 31:
+                            return "Engine " + A + " - Blow Horn 2";
+                        default:
+                            return "Engine " + A + " - action command D=" + D;
+                    }
+
+                case 1:
+                    //return "Engine " + A + " - extended command (C=1) with D=" + D;
+                    if ((D & 0x17) == 0) {
+                        return "Engine " + A + " - Momentum Low";
+                    }
+                    if ((D & 0x17) == 1) {
+                        return "Engine " + A + " - Momentum Medium";
+                    }
+                    if ((D & 0x17) == 2) {
+                        return "Engine " + A + " - Momentum High";
+                    }
+                    if ((D & 0x17) == 3) {
+                        return "Engine " + A + " - Set";
+                    }
+                    if ((D & 0x17) == 5) {
+                        return "Engine " + A + " - Start Up Sequence 1 (Delayed Prime Mover, then Immediate Start Up)";
+                    }
+                    if ((D & 0x17) == 6) {
+                        return "Engine " + A + " - Start Up Sequence 2 (Immediate Start Up)";
+                    }
+                    if ((D & 0x17) == 7) {
+                        return "Engine " + A + " - Shut Down Sequence 1 (Delay w/ Announcement then Immediate Shut Down)";
+                    }
+                    if ((D & 0x17) == 8) {
```

Comment:
The expression `((D & 0x17) == 8)` will always be `false`.

What's the intention of this line? Do you want to check if the lowest 5 bits has the value of 8? If so, the expression should be:
`((D & 0x1F) == 8)`

But if you want to check if the lowest 4 bits has the value of 8, the expression should be:
`((D & 0x0F) == 8)`